mifosng-provider/src/main/java/org/mifosplatform/infrastructure/core/domain/MifosPlatformTenant.java

```diff
@@ -133,36 +41,24 @@ public String getName() {
         return this.name;
     }
 
-    public String getSchemaName() {
-        return this.schemaName;
-    }
-
-    public String getSchemaUsername() {
-        return this.schemaUsername;
-    }
-
-    public String getSchemaPassword() {
-        return this.schemaPassword;
-    }
-
     public String getTimezoneId() {
         return this.timezoneId;
     }
 
-    public boolean isAutoUpdateEnabled() {
-        return this.autoUpdateEnabled;
+    public MifosPlatformTenantConnection getOltpConnection() {
+        return oltpConnection;
     }
 
-    public int getSuspectTimeout() {
-        return this.suspectTimeout;
+    public MifosPlatformTenantConnection getReportsConnection() {
+        return reportsConnection;
     }
 
-    public int getMaxRetriesOnDeadlock() {
-        return maxRetriesOnDeadlock;
+    public Long getOltpId() {
+        return oltpId;
     }
 
-    public int getMaxIntervalBetweenRetries() {
-        return maxIntervalBetweenRetries;
+    public Long getReportId() {
+        return reportId;
     }
 
```

Comment:
Are you suggesting basicAuthTenantDetailsService.loadTenantById(tenantIdentifier) service API to be modified to take additional parameter to indicate if the request is for reports? So that OLTP/Reports decision is encapsulated within MifosPlatformTenant class?